core/src/main/java/org/infinispan/interceptors/locking/OptimisticLockingInterceptor.java

```diff
@@ -39,68 +30,27 @@
  */
 public class OptimisticLockingInterceptor extends AbstractTxLockingInterceptor {
 
-   private LockAcquisitionVisitor lockAcquisitionVisitor;
-   private boolean needToMarkReads;
-
    private static final Log log = LogFactory.getLog(OptimisticLockingInterceptor.class);
-
-   @Override
-   protected Log getLog() {
-      return log;
-   }
+   private boolean needToMarkReads;
 
    @Start
    public void start() {
-      if (cacheConfiguration.clustering().cacheMode() == CacheMode.LOCAL &&
+      needToMarkReads = cacheConfiguration.clustering().cacheMode() == CacheMode.LOCAL &&
```

Comment:
because in clustered mode we use the versions to know when the key is read. So, you don't need to mark the reads.

In local mode, we don't use any versioning, so we need to keep track of the keys read to perform the write skew check.

About the unsafe, it is before my time. But I think we need it because it affects the writes (remember that the put() should read the key before writing it (except if we have flags))